### PR TITLE
Update to 7.58.0

### PR DESCRIPTION
**Pull request by ocefpaf**
Just testing...

### CODE REVIEW

**Comment by ocefpaf** [comment]
@nehaljwani I noticed that the tests are disabled for the anaconda version of this recipe. What are the reasons for doing so? Same failures we are seeing here?

**Comment by nehaljwani** [comment]
If I understand your query correctly, the tests for forked recipes are run in the internal CI system and then the changes are pushed directly. They are not tested on CircleCI/Appveyor/Travis.

**Comment by ocefpaf** [comment]
> If I understand your query correctly, the tests for forked recipes are run in the internal CI system and then the changes are pushed directly. They are not tested on CircleCI/Appveyor/Travis.

Actually I was referring to this line: https://github.com/AnacondaRecipes/curl-feedstock/blob/master/recipe/build.sh#L17

**Comment by nehaljwani** [comment]
@mingwandroid did that. There was probably a failing test which could be ignored.

**Comment by ocefpaf** [comment]
In our case we `2056` and `2057` failing on OS X :unamused:

**Comment by ocefpaf** [comment]
Merging this without OS X to get the other changes in #27

File: recipe/meta.yaml [hunk 1]
```diff
@@ -45,6 +45,8 @@ test:
     - curl https://raw.githubusercontent.com/conda-forge/curl-feedstock/master/LICENSE
     - conda inspect linkages -p $PREFIX $PKG_NAME  # [not win]
     - conda inspect objects -p $PREFIX $PKG_NAME  # [osx]
+    - nm $PREFIX/lib/libcurl.so | grep libssh2_scp_recv2  # [not win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Also while `nm` does exist. `.so` is really only on Linux (except for some Python extensions). Should be `$SHLIB_EXT` instead.

**Comment by jakirkham** [comment]
The tests also fail upstream [as noted in this comment]( https://github.com/curl/curl/pull/1975#issuecomment-335436563 ). Think patching them out on macOS is pretty defensible. Hopefully that PR lands soon so these will be fixed.

**Pull request merged**